rosbag2_cpp/test/rosbag2_cpp/test_time_controller_clock.cpp

```diff
@@ -47,86 +47,94 @@ TEST_F(TimeControllerClockTest, must_provide_now_fn)
 {
   NowFunction empty_now;
   EXPECT_THROW(
-    rosbag2_cpp::TimeControllerClock(ros_start_time, 1.0, empty_now),
+    rosbag2_cpp::TimeControllerClock(ros_start_time, empty_now),
     std::invalid_argument);
 }
 
 TEST_F(TimeControllerClockTest, steadytime_precision)
 {
-  const double playback_rate = 1.0;
-  rosbag2_cpp::TimeControllerClock pclock(ros_start_time, playback_rate, now_fn);
+  rosbag2_cpp::TimeControllerClock clock(ros_start_time, now_fn);
 
   const SteadyTimePoint begin_time(std::chrono::seconds(0));
   return_time = begin_time;
-  EXPECT_EQ(pclock.now(), as_nanos(begin_time));
+  EXPECT_EQ(clock.now(), as_nanos(begin_time));
 
   const SteadyTimePoint ten_seconds(std::chrono::seconds(10));
   return_time = ten_seconds;
-  EXPECT_EQ(pclock.now(), as_nanos(ten_seconds));
+  EXPECT_EQ(clock.now(), as_nanos(ten_seconds));
 
   // NOTE: this would have already lost precision at 100 seconds if we were multiplying by float
   const SteadyTimePoint hundred_seconds(std::chrono::seconds(100));
   return_time = hundred_seconds;
-  EXPECT_EQ(pclock.now(), as_nanos(hundred_seconds));
+  EXPECT_EQ(clock.now(), as_nanos(hundred_seconds));
 
   const int64_t near_limit_nanos = 1LL << 61;
   const auto near_limit_time = SteadyTimePoint(std::chrono::nanoseconds(near_limit_nanos));
   return_time = near_limit_time;
-  EXPECT_EQ(pclock.now(), near_limit_nanos);
+  EXPECT_EQ(clock.now(), near_limit_nanos);
 
   // Expect to lose exact equality at this point due to precision limit of double*int mult
   const int64_t over_limit_nanos = (1LL << 61) + 1LL;
   const auto over_limit_time = SteadyTimePoint(std::chrono::nanoseconds(over_limit_nanos));
   return_time = over_limit_time;
-  EXPECT_NE(pclock.now(), over_limit_nanos);
-  EXPECT_LT(std::abs(pclock.now() - over_limit_nanos), 10LL);
+  EXPECT_NE(clock.now(), over_limit_nanos);
+  EXPECT_LT(std::abs(clock.now() - over_limit_nanos), 10LL);
 }
 
 TEST_F(TimeControllerClockTest, nonzero_start_time)
 {
   ros_start_time = 1234567890LL;
-  const double playback_rate = 1.0;
-  rosbag2_cpp::TimeControllerClock pclock(ros_start_time, playback_rate, now_fn);
+  rosbag2_cpp::TimeControllerClock clock(ros_start_time, now_fn);
+  clock.set_rate(1.0);
```

Comment:
I would add 
```cpp
EXPECT_TRUE(clock.set_rate(1.0)); 
```
And somewhere 
```cpp
EXPECT_FALSE(clock.set_rate(0.0));
EXPECT_FALSE(clock.set_rate(-0.5));
```

Reply:
I think that's done further down below, within "invalid rate"

Reply:
Yep, it's a separate test

Reply:
Sorry I overlooked it.